src/api/java/com/minecolonies/api/crafting/RecipeStorage.java

```diff
@@ -186,6 +191,10 @@ public RecipeStorage(final IToken<?> token, final List<ItemStack> input, final i
         {
             if (ItemStackUtils.isEmpty(stack) || stack.getItem() == ModItems.buildTool)
             {
+                if(stack.getItem() == ModItems.buildTool)
+                {
+                    usesBuildTool = true;
```

Comment:
While tracking this isn't critical, it allows the "full" input check during the custom recipe check without falling back to the uncleaned inputs and recleaning them each time. It also can be ordered  first in the evaluation, and becomes a great short-circuit for the way we use prereq recipes today (upgrading a normal to a multi-output) since all of our recipes today use a build tool in the inputs.

Reply:
If we remove the usesBuildTool tracking, and leave the additional getPrimaryOutput comparison, it solves the observed problems, but leaves us with a possibility of inputs matching when only one side has the build tool.